core/blockchain.ex

```diff
@@ -27,4 +30,24 @@ defmodule UltraDark.Blockchain do
     chain
   end
 
+  def recalculate_difficulty(chain) do
+    beginning_index = min(length(chain) - 1, @diff_rebalance_offset - 1)
+
+    last = List.first(chain)
+    first = Enum.at(chain, beginning_index)
+
+    diff =
+      with {:ok, last_time, _} <- DateTime.from_iso8601(last.timestamp),
+           {:ok, first_time, _} <- DateTime.from_iso8601(first.timestamp) do
+        diff_µs = DateTime.diff(last_time, first_time, :microseconds)
```

Comment:
lets not use special characters here, it makes it hard to write code that references this variable without copy+pasting